### PR TITLE
chore: conditional helmfile commands on pull request

### PR DESCRIPTION
**Pull request by lemeurherve**
Lint and diff on pull requests only if the current cluster helmfile or any of the chart configurations files have changed

Revert https://github.com/jenkins-infra/kubernetes-management/pull/2790

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -100,13 +100,44 @@ pipeline {
...
+            // Diff on pull requests only if the current cluster helmfile or any of the chart configurations files have changed
             when {
-              changeRequest()
+              allOf {
+                changeRequest()
+                anyOf {
+                  changeset "config/**"
+                  expression {
+                    sh(returnStatus: true, script: '''
+                      git status -s | grep "^clusters/${K8S_CLUSTER}.yaml"
```
Comment (halkeye) [inline Jenkinsfile_k8s:135]: doesn't changeset let you do that?

Comment (halkeye) [reply Jenkinsfile_k8s:135]: `when { changeset pattern: "^clusters/${K8S_CLUSTER}.yaml", comparator: "REGEXP" }`

Comment (lemeurherve) [reply Jenkinsfile_k8s:135]: I've tried before with `changeset "clusters/${K8S_CLUSTER}.yaml" but it didn't worked. (Tested again with a replay here: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/3/console)

I've just tried your syntax in another replay but it doesn't seem to work either: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/4/console

Unfortunately, git+grep doesn't seem to work too: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/5/console

Comment (halkeye) [reply Jenkinsfile_k8s:135]: well grep by default isn't using regex, so you want egrep or grep -e

I'll try to take a look at those links when i'm on a computer i can run vpn again.

**Comment by dduportal** [comment]
I wonder if the (code) complexity / value is still interesting: it adds a lot of (untested -because it's a pipeline) code paths without speeding up the overall result (the 4 clusters are diff-ed in parallel).

What improvements did you see in your experiments?

**Comment by lemeurherve** [comment]
> I wonder if the (code) complexity / value is still interesting

Probably not, closing.

**Pull request closed without merging**